lib/ProMotion/table/table_builder.rb

```diff
@@ -1,13 +1,24 @@
 module ProMotion
   module TableBuilder
     def trigger_action(action, arguments, index_path)
-      return mp("Action not implemented: #{action.to_s}", force_color: :green) unless self.respond_to?(action)
-      case arity = self.method(action).arity
-      when 0 then self.send(action) # Just call the method
-      when 2 then self.send(action, arguments, index_path) # Send arguments and index path
-      else 
-        mp("Action should not have optional parameters: #{action.to_s}", force_color: :yellow) if arity < 0
-        self.send(action, arguments) # Send arguments
+      if action.is_a?(Proc)
```

Comment:
I think it would be better to not duplicate code here. Resolve to a Method or Proc first.

```ruby
action = (action.is_a?(Proc) ? action : method(action))
case arity = action.arity
when # ...
```